File: HRMS/HRMS.Web/Modules/ServerTypes/HR/DepartmentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DepartmentRow } from "./DepartmentRow";

export interface DepartmentColumns {
    DepartmentId: Column<DepartmentRow>;
    DepartmentName: Column<DepartmentRow>;
    Description: Column<DepartmentRow>;
    IsActive: Column<DepartmentRow>;
}

export class DepartmentColumns extends ColumnsBase<DepartmentRow> {
    static readonly columnsKey = 'HR.Department';
    static readonly Fields = fieldsProxy<DepartmentColumns>();
}

File: HRMS/HRMS.Web/Modules/HR/Department/DepartmentColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.HR.Columns;

[ColumnsScript("HR.Department")]
[BasedOnRow(typeof(DepartmentRow), CheckNames = true)]
public class DepartmentColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DepartmentId { get; set; }
    [EditLink]
    public string DepartmentName { get; set; }
    public string Description { get; set; }
    [Width(100), AlignCenter]
    public bool IsActive { get; set; }
}
